matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/summary/RoomSummaryDataSource.kt

```diff
@@ -230,6 +239,16 @@ internal class RoomSummaryDataSource @Inject constructor(
         }
     }
 
+    fun getCountFlow(queryParams: RoomSummaryQueryParams): Flow<Int> =
+            realmSessionProvider
+                    .withRealm { realm -> roomSummariesQuery(realm, queryParams).findAllAsync() }
+                    .toFlow()
+                    // need to create the flow on a context dispatcher with a thread with attached Looper
+                    .flowOn(coroutineDispatchers.main)
```

Comment:
this is a little bit strange, we need to switch to the main thread in order to query the count?  

we have some other flow based pipelines we don't need to do the switch https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android-flow/src/main/java/org/matrix/android/sdk/flow/FlowSession.kt :thinking:

Reply:
Yes, I see why it could be strange but this is the only solution I found to make it work. I am using the [toFlow() extension of RealmResult](https://github.com/realm/realm-java/blob/973d43725d3bc2a8bd053892ddab0ff2856daa12/realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmResultsExtensions.kt) and I didn't find any existing usage in the project. I have seen there is an [issue about the documentation of this extension](https://github.com/realm/realm-java/issues/7168) meaning the thread we collect the flow should be the same as the one used for creating the result...

Reply:
having read a little more into this, it's not as troublesome as I originally thought, realm's async is keeping us off of the main thread but it needs to be triggered from a thread with a looper

initially I was thought this might have pushed us into doing the database query on the main thread